src/circuits/circuits.jl

```diff
@@ -131,30 +135,97 @@ followed by a layer of single qubit gates. By default, the two-qubit gate
 is controlled-NOT, and the single-qubit gate is a rotation around a random
 axis. 
 """
-function randomcircuit(N::Int,depth::Int,twoqubit_bonds::Array;
-                       twoqubitgate   = "CX",
-                       onequbitgates  = ["Rn"])
-  gates = Tuple[]
+function randomcircuit(N::Int, depth::Int, twoqubit_bonds::Array;
+                       twoqubitgate::String   = "randU",
+                       onequbitgates::Array  = [],
+                       layered::Bool = true,
+                       seed = nothing)
```

Comment:
So here, I'm now not sure if passing a `seed` is the best design. I think it is better to use an RNG here, since a user might want to do something like:
```julia
rng = MersenneTwister(1234)
circuit1 = randomcircuit(4, 2; rng = rng)
circuit2 = randomcircuit(4, 2; rng = rng) # This continues on with the same rng stream
# This repeats the same sequence of gates:
Random.seed!(rng, 1234)
circuit1p = randomcircuit(4, 2; rng = rng)
circuit2p = randomcircuit(4, 2; rng = rng) # This continues on with the same rng stream
```
This seems like a reasonable thing to want to do (if you are making multiple random circuits, you probably don't want to have to come up with multiples seeds for each one).

Like in the comment I made for the `twoqubitlayer!` function, I think that we should design it so that the `rng` gets passed right to the gates, for example:
```julia
g = ("Rn", j, (rng = rng,))
g = ("randU", j, (rng = rng,))
```
and we can change the definitions of the gates `"Rn"` and `"randU"` to accept an RNG which is then used to generate random angles (in the case of `"Rn"`) or a random matrix (in the case of `"randU"`). This seems like a more "generic" solution, and lets people easily create random gates and circuits outside of the context of the `randomcircuit` function. For example, one could do:
```julia
N = 10
psi = quibits(N)
rng = MersenneTwister(1234)
gates = [("Rn", j, (rng = rng,)) for j in 1:N]
# Make two random layers
layer1 = buildcircuit(psi, gates)
layer2 = buildcircuit(psi, gates)
# This would make the same set of random layers
Random.seed!(rng, 1234)
layer1p = buildcircuit(psi, gates)
layer2p = buildcircuit(psi, gates)
```
Let me know if this makes sense.